lib/srv/db/objects/objects.go

```diff
@@ -0,0 +1,208 @@
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package objects
+
+import (
+	"context"
+	"log/slog"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/client/databaseobject"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/cloud"
+	"github.com/gravitational/teleport/lib/srv/db/common"
+)
+
+type Objects interface {
+	StartImporter(ctx context.Context, database types.Database) error
+	StopImporter(databaseName string) error
+}
+
+type Config struct {
+	DatabaseObjectClient *databaseobject.Client
+	ImportRules          ImportRulesReader
+	Auth                 common.Auth
+	CloudClients         cloud.Clients
+
+	// ScanInterval specifies how often the database is scanned.
+	// A higher ScanInterval reduces the load on the database and database agent,
+	// but increases the delay in detecting schema changes or updates from new import rules.
+	ScanInterval time.Duration
+
+	// ObjectTTL defines TTL for a newly created object.
+	// A higher ObjectTTL reduces the backend load and should be significantly larger than ScanInterval.
+	// Setting a TTL for database objects ensures cleanup if the database becomes unavailable or the database agent stops performing scans.
+	ObjectTTL time.Duration
+
+	// RefreshThreshold sets the minimum remaining TTL for an object to qualify for a TTL refresh.
+	RefreshThreshold time.Duration
+
+	Clock clockwork.Clock
+	Log   *slog.Logger
+}
+
+func (c *Config) loadEnvVar(ctx context.Context, name string) (bool, time.Duration) {
+	envVar := os.Getenv(name)
+	if envVar == "" {
+		return false, 0
+	}
+	if envVar == "never" {
+		return true, 0
+	}
+
+	interval, err := time.ParseDuration(envVar)
+	if err != nil {
+		c.Log.ErrorContext(ctx, "Failed to parse env var, override not applied.", "name", name, "value", envVar)
+	}
+
+	return true, interval
+}
+
+func (c *Config) loadEnvVarOverrides(ctx context.Context) {
+	needInfo := false
+	// overriding scan interval modifies the other variables, but not the other way around.
+	if found, value := c.loadEnvVar(ctx, "TELEPORT_UNSTABLE_DB_OBJECTS_SCAN_INTERVAL"); found {
+		c.ScanInterval = value
+		c.ObjectTTL = value * 12
+		c.RefreshThreshold = value * 3
```

Comment:
could you add a comment explaining why the numbers 12 and 3?
It may be a good idea also to add the unit of time to these env vars to make it clear, e.g. `TELEPORT_UNSTABLE_DB_OBJECTS_OBJECT_TTL_MINUTES` (I'm actually not sure what the units are assumed to be currently)

edit: ok I see `time.ParseDuration` handles a wide range of time formats, including negative values.
I think I still prefer something like `_MINUTES` and just parsing an unsigned int, with a check for the 0 value if 0 is invalid. That enforces a minimum value of 1 minute automatically, and is simpler to reason about plus the env var name documents itself.

Reply:
I think I picked this pattern from elsewhere in Teleport, feels good to keep to one convention. The updated logic checks for zero/negative intervals which matter, so we should be good. In any case messing with these `UNSTABLE` env vars puts responsibility on the administrator to pick sensible values and monitor the stability afterwards.